src/map/status.c

```diff
@@ -12409,8 +12418,10 @@ int status_natural_heal(struct block_list* bl, va_list args) {
 		}
 	}
 
-	if (flag && regen->state.overweight)
-		flag=0;
+	// SC_TENSIONRELAX allows HP to be recovered even when overweight. [csnv]
+	if (flag && regen->state.overweight && (sc && !sc->data[SC_TENSIONRELAX])) {
```

Comment:
Did you mean `if (flag && regen->state.overweight && (sc == NULL || sc->data[SC_TENSIONRELAX] == NULL))`? (if sc is `NULL`, it means that `SC_TENSIONRELAX` is inactive, so it should enter the if)